include/OpenColorIO/OpenColorIO.h

```diff
@@ -3110,19 +3114,43 @@ class OCIOEXPORT GpuShaderCreator
         TEXTURE_RGB_CHANNEL  ///< Need a RGB texture
     };
 
+    /**
+     * Dimension enum used to differentiate between 1D and 2D object/resource types.
+     */
+    enum TextureDimensions : uint8_t {
+        TEXTURE_1D = 1,
+        TEXTURE_2D = 2,
+    };
+
     /**
      *  Add a 2D texture (1D texture if height equals 1).
      * 
      * \note 
      *   The 'values' parameter contains the LUT data which must be used as-is as the dimensions and
      *   origin are hard-coded in the fragment shader program. So, it means one GPU texture per entry.
      **/
-    virtual void addTexture(const char * textureName,
-                            const char * samplerName,
+    OCIO_DEPRECATED("This was marked as deprecated starting in v2.3, please use addTexture() with dimensions.")
+    virtual void addTexture(const char* /*textureName*/,
+                            const char* /*samplerName*/,
+                            unsigned /*width*/, unsigned /*height*/,
+                            TextureType /*channel*/,
+                            Interpolation /*interpolation*/,
+                            const float* /*values*/) {}
```

Comment:
Why having an empty default implementation? 
The implementation could redirect to the new method with `dimensions` equal 2D. By doing so, the implementation remains correct.

Reply:
Thanks for the notes, will update! Although I suppose I should then remove the overrides in `GenericGpuShaderDesc`.

I would also expect `dimensions` to be `const TextureDimensions dimensions = height > 1 ? TextureDimensions::TEXTURE_2D : TextureDimensions::TEXTURE_1D;` instead of always 2D, since this would be the backward-compatible behavior.

Reply:
I was wondering why you didn't use the `getAllowTexture1D()` is those implementations, is that to preserve old behaviour?

Reply:
To mimic the old behavior yes, `height > 1` was previously the implicit criteria to decide which whether the resource was 1D or 2D.